### PR TITLE
Added support for loading oneofs from descriptors.

### PR DESCRIPTION
**Pull request by haberman**
Review to @scwhittle. cc @teboring.

### CODE REVIEW

File: upb/descriptor/reader.c [hunk 1]
```diff
@@ -145,6 +159,23 @@ void upb_descreader_setscopename(upb_descreader *r, char *str) {
...
 
+static upb_oneofdef *upb_descreader_getoneof(upb_descreader *r,
+                                             uint32_t index) {
+  bool found;
+  upb_value val;
+  upb_descreader_frame *f = &r->stack[r->stack_len-1];
+
+  index += f->oneof_start;
+
+  while (upb_inttable_count(&r->oneofs) <= index) {
```
Comment (scwhittle) [inline upb/descriptor/reader.c]: this seems a little odd, why do you need all the in-between ones?

File: upb/descriptor/reader.c [hunk 2]
```diff
@@ -536,6 +567,35 @@ static size_t field_ondefaultval(void *closure, const void *hd, const char *buf,
   return n;
 }
 
+static bool field_ononeofindex(void *closure, const void *hd, int32_t index) {
+  upb_descreader *r = closure;
+  upb_oneofdef *o = upb_descreader_getoneof(r, index);
+  bool ok;
```
Comment (scwhittle) [inline upb/descriptor/reader.c]: bool ok = ....

File: upb/descriptor/reader.c [hunk 3]
```diff
@@ -536,6 +567,35 @@ static size_t field_ondefaultval(void *closure, const void *hd, const char *buf,
...
+
+/** Handlers for google.protobuf.OneofDescriptorProto. ************************/
+
+static size_t oneof_name(void *closure, const void *hd, const char *buf,
+                         size_t n, const upb_bufhandle *handle) {
+  upb_descreader *r = closure;
+  upb_descreader_frame *f = &r->stack[r->stack_len-1];
+  upb_oneofdef *o = upb_descreader_getoneof(r, f->oneof_index++);
+  bool ok;
+  char *name_null_terminated = upb_strndup(buf, n);
```
Comment (scwhittle) [inline upb/descriptor/reader.c]: ditto, can you move ok beneath name_null_terminated and assign directly?

File: upb/descriptor/reader.c (hunk 2 above)
Comment (haberman) [inline upb/descriptor/reader.c]: Done.

File: upb/descriptor/reader.c (hunk 3 above)
Comment (haberman) [inline upb/descriptor/reader.c]: Done.

**Pull request merged**